File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++ ) {
            map.put(nums[i], i);
        }
        for(int i = 0; i < nums.length; i++) {
            int num2 = target - nums[i];
            
            
                if (map.containsKey(num2)) {
                    int otherIndex = map.get(num2);
                    if (otherIndex != i ) {
                    int[] result = {i , map.get(num2)};
                return result;
            }
            }
            
            
        }
        return null;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l , r = 0, len(nums) - 1 
        if nums[l] < nums[r]:
            return nums[l]
        while r - l > 1:

            c = l + (r - l )// 2
        
            if nums[l] > nums[c]:
                r = c
            elif nums[c] > nums[r]:
                l = c 
        
        
        return min(nums[l], nums[r])
